### PR TITLE
ping: init 0.6.0

### PR DESCRIPTION
**Pull request by xiorcale**
###### Motivation for this change

Since Pantheon desktop has recently been added to Master, adding some "curated" apps from Elementary AppCenter seems to be a good idea to complete it.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: maintainers/maintainer-list.nix
Comment (worldofpeace) [inline maintainers/maintainer-list.nix]:
```suggestion
```

File: pkgs/development/web/ping/default.nix
Comment (worldofpeace) [inline pkgs/development/web/ping/default.nix]:
```suggestion
  pname = "ping";
```

File: pkgs/development/web/ping/default.nix
Comment (worldofpeace) [inline pkgs/development/web/ping/default.nix]:
```suggestion
    repo = pname;
```

File: pkgs/development/web/ping/default.nix
Comment (worldofpeace) [reply pkgs/development/web/ping/default.nix]: btw `name` defaults to `${pname}-${version}` when this attribute is present.

File: pkgs/development/web/ping/default.nix
Comment (worldofpeace) [inline pkgs/development/web/ping/default.nix]:
```suggestion
    description = "A helpful tool that lets you degug what part of your API is causing you issues";
```

File: pkgs/development/web/ping/default.nix
Comment (worldofpeace) [inline pkgs/development/web/ping/default.nix]:
```suggestion
    maintainers = with maintainers; [ kjuvi ] ++ pantheon.maintainers;
```

Useful for me since I maintain the underlying packaging set.

File: pkgs/top-level/all-packages.nix
Comment (worldofpeace) [inline pkgs/top-level/all-packages.nix]:
```suggestion
  ping = callPackage ../development/web/ping { };
```

Comment (worldofpeace) [reply pkgs/top-level/all-packages.nix]: Also I don't think this belongs in `development/web`.

This is a gui application that is useful for debugging web API's, so maybe `applications/networking`.

File: pkgs/development/web/ping/default.nix
Comment (worldofpeace) [inline pkgs/development/web/ping/default.nix]:
```suggestion
    glib
    gtk3
```

**Comment by worldofpeace** [comment]
Can you also pull adding yourself to the maintainers-list into a separate commit?

```
maintainers: add kjuwi
```

File: pkgs/development/web/ping/default.nix
Comment (worldofpeace) [inline pkgs/development/web/ping/default.nix]:
```suggestion
    gtksourceview
    hicolor-icon-theme
```

File: pkgs/development/web/ping/default.nix
Comment (Kjuvi) [reply pkgs/development/web/ping/default.nix]: Oh, I didn't know the difference between ```name``` and ```pname```, thanks !

File: pkgs/top-level/all-packages.nix
Comment (Kjuvi) [reply pkgs/top-level/all-packages.nix]: I put it in ```development/web``` because ```postman``` was there, which is more or less the same application than ```ping```

Furthermore, it's a tool used for debugging API which are under development, so I think it makes sense to keep it here.

Comment (worldofpeace) [reply pkgs/top-level/all-packages.nix]: Manual says that for things under `development` should be
> it’s used to support software development

And then goes on to mention libraries, compilers, and build-managers.
None on which will ever become a gui application.

You could say that an application like this would support software development, but so could an IDE, and they never should go under `development`.

I'd say that things like `postman` should be moved to a new place under `applications/` and the closest thing we have currently is something like `applications/networking`.

Comment (Kjuvi) [reply pkgs/top-level/all-packages.nix]: I totally agree with you, I actually wanted to put the app under ```applications/networking``` before I discover that ```postman``` was under ```development/web```

What should I do then ? Keeping consistency with the actual organization or move ```ping``` under ```applications/networking``` to better agree with the manual ?

Comment (worldofpeace) [reply pkgs/top-level/all-packages.nix]: I'd say `applications/networking` would be the best place currently.

We usually clean up the actual organization eventually.

Comment (Kjuvi) [reply pkgs/top-level/all-packages.nix]: Alright, I will move it to the right place then!

Also, I was wondering, wouldn't it be more convenient to have the curated apps from Elementary App Center organized under ```desktop/pantheon``` in a special directory for them? Because some of these apps seem to be tightly coupled with some part of the desktop (e.g. Wingpanel, ...).

Comment (worldofpeace) [reply pkgs/top-level/all-packages.nix]: > Alright, I will move it to the right place then!

Exclamation here confused me.

> Also, I was wondering, wouldn't it be more convenient to have the curated apps from Elementary App Center organized under desktop/pantheon in a special directory for them? Because some of these apps seem to be tightly coupled with some part of the desktop (e.g. Wingpanel, ...).

I thought of this too, and I've decided that apps that specifically won't function without some part of the pantheon desktop can be organized there.  
I've  also patched [granite](https://github.com/NixOS/nixpkgs/blob/32767d139f28fd3c00d687c04ec406258f7341e7/pkgs/desktops/pantheon/granite/default.nix#L15) so that they work outside pantheon (not needing wingpanel).
And even if they don't work I'd like them to default to not needing their desktop.
See: https://github.com/NixOS/nixpkgs/commit/ac2546bffe6c980642c9f6a71df06efa226b8816

File: pkgs/top-level/all-packages.nix
Comment (worldofpeace) [inline pkgs/top-level/all-packages.nix]: Also remove the added whitespace here when you cleanup the history.

File: pkgs/top-level/all-packages.nix
Comment (worldofpeace) [inline pkgs/top-level/all-packages.nix]: Remove the whitespace here as well.

File: pkgs/top-level/all-packages.nix
Comment (worldofpeace) [inline pkgs/top-level/all-packages.nix]: Please remove the trailing whitespace here when you cleanup the history.

File: pkgs/top-level/all-packages.nix
Comment (worldofpeace) [inline pkgs/top-level/all-packages.nix]: Trailing whitespace here as well.

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Everything's good, and I'll merge once the history is cleaned up to just having the two commits.
```
maintainers: add kjuvi
ping: init at 0.6.0
```

File: pkgs/top-level/all-packages.nix
Comment (Kjuvi) [reply pkgs/top-level/all-packages.nix]: > Exclamation here confused me.

well, I was maybe a bit too enthusiastic that we found an agreement :smile:

I had no idea about granite, but it makes sense yes. And the idea of the flag to default to not needing their desktop seems convenient too. Thanks for the details.

File: pkgs/top-level/all-packages.nix
Comment (Kjuvi) [reply pkgs/top-level/all-packages.nix]: I did clean the history, but I'm not sure I did it correctly (I'm not used to squash commits...). Can you confirm that everything is good ?

File: pkgs/top-level/all-packages.nix
Comment (Kjuvi) [reply pkgs/top-level/all-packages.nix]: I did clean the history, but I'm not sure I did it correctly (I'm not used to squash commits...). Can you confirm that everything is good ?

**Review by worldofpeace — APPROVED** [review]

**Comment by worldofpeace** [comment]
Hope you found the review informative @Kjuvi , and thank you for your contribution :sparkles:

**Pull request merged**
